src/gt4py/next/program_processors/runners/dace_fieldview/transformations/map_fusion_serial.py

```diff
@@ -0,0 +1,1004 @@
+            access_set: Set[nodes.AccessNode] = self.get_access_set(scope_node, state)
+            access_sets.append({node.data: node for node in access_set})
+            # If two different access nodes of the same scoping node refers to the
+            #  same data, then we consider this as a dependency we can not handle.
+            #  It is only a problem for the intermediate nodes and might be possible
+            #  to handle, but doing so is hard, so we just forbid it.
+            if len(access_set) != len(access_sets[-1]):
+                return True
+        read_map_1, write_map_1, read_map_2, write_map_2 = access_sets
+
+        # It might be possible that there are views, so we have to resolve them.
+        #  We also already get the name of the data container.
+        #  Note that `len(real_read_map_1) <= len(read_map_1)` holds because of Views.
+        resolved_sets: List[Set[str]] = []
+        for unresolved_set in [read_map_1, write_map_1, read_map_2, write_map_2]:
+            resolved_sets.append(
+                {
+                    self.track_view(node, state, sdfg).data
+                    if self.is_view(node, sdfg)
+                    else node.data
+                    for node in unresolved_set.values()
+                }
+            )
+            # If the resolved and unresolved names do not have the same length.
+            #  Then different views point to the same location, which we forbid
+            if len(unresolved_set) != len(resolved_sets[-1]):
+                return False
+        real_read_map_1, real_write_map_1, real_read_map_2, real_write_map_2 = resolved_sets
+
+        # We do not allow that the first and second map each write to the same data.
+        if not real_write_map_1.isdisjoint(real_write_map_2):
+            return True
+
+        # If there is no overlap in what is (totally) read and written, there will be no conflict.
+        #  This must come before the check of disjoint write.
+        if (real_read_map_1 | real_read_map_2).isdisjoint(real_write_map_1 | real_write_map_2):
+            return False
+
+        # These are the names (unresolved) and the access nodes of the data that is used
+        #  to transmit information between the maps. The partition function ensures that
+        #  these nodes are directly connected to the two maps.
+        exchange_names: Set[str] = set(write_map_1.keys()).intersection(read_map_2.keys())
+        exchange_nodes: Set[nodes.AccessNode] = set(write_map_1.values()).intersection(
+            read_map_2.values()
+        )
+
+        # If the number are different then a data is accessed through multiple nodes.
+        if len(exchange_names) != len(exchange_nodes):
+            return True
+        assert all(exchange_node.data in exchange_names for exchange_node in exchange_nodes)
+
+        # For simplicity we assume that the nodes used for exchange are not views.
```

Comment:
The intermediate does not need to be a transient, in certain cases it can also be global memory.
This is essentially done in `is_shared_data()` which is called through the partition function.